python/ray/serve/_private/config.py

```diff
@@ -170,6 +171,51 @@ class DeploymentConfig(BaseModel):
     # Contains the names of deployment options manually set by the user
     user_configured_option_names: Set[str] = set()
 
+    @staticmethod
+    def _get_single_max_batch_size_from_method(method: Callable) -> int:
+        """Helper to get a single max_batch_size form a method.
+
+        If the method doesn't uses batching, return 0.
+        """
+        if not hasattr(method, "_get_max_batch_size"):
+            return 0
+
+        return method._get_max_batch_size()
+
+    def check_max_batch_size_bounded(
```

Comment:
I think it makes more sense to this inside of `serve.batching._BatchQueue` rather than adding this method parsing hackery. As written, this currently does not work if users use the `set_max_batch_size` setter on the method (which can be done dynamically at runtime in the replica).

We'll need access to `max_ongoing_requests` there, for which I'd suggest we add `deployment_config` as a private attribute of `ReplicaContext` (and set/update it properly in `replica.py` for that purpose). This will likely be generally useful in the future anyways.

Reply:
Oh cool, didn't know there is a way for batching to get `max_ongoing_requests`. Thanks for the suggestion Ed!